packages/access-api/src/service/access-claim.js

```diff
@@ -0,0 +1,53 @@
+import * as Server from '@ucanto/server'
+import { claim } from '@web3-storage/capabilities/access'
+import * as Ucanto from '@ucanto/interface'
+import { collect } from 'streaming-iterables'
+import * as delegationsResponse from '../utils/delegations-response.js'
+
+/**
+ * @typedef AccessClaimSuccess
+ * @property {Record<string,Ucanto.ByteView<Ucanto.Delegation>>} delegations
+ */
+
+/**
+ * @typedef {{ error: true }} AccessClaimFailure
+ * @typedef {Ucanto.Result<AccessClaimSuccess, AccessClaimFailure>} AccessClaimResult
+ */
+
+/**
+ * @param {object} ctx
+ * @param {import('../types/delegations').DelegationsStorage} ctx.delegations
+ * @param {Pick<import('../bindings.js').RouteContext['config'], 'ENV'>} ctx.config
+ */
+export function accessClaimProvider(ctx) {
+  const handleClaimInvocation = createAccessClaimHandler(ctx)
+  return Server.provide(claim, async ({ invocation }) => {
+    // disable until hardened in test/staging
+    if (ctx.config.ENV === 'production') {
+      throw new Error(`acccess/claim invocation handling is not enabled`)
+    }
+    return handleClaimInvocation(invocation)
+  })
+}
+
+/**
+ * @callback AccessClaimHandler
+ * @param {Ucanto.Invocation<import('@web3-storage/capabilities/types').AccessClaim>} invocation
+ * @returns {Promise<AccessClaimResult>}
+ */
+
+/**
+ * @param {object} options
+ * @param {import('../types/delegations').DelegationsStorage} options.delegations
+ * @returns {AccessClaimHandler}
+ */
+export function createAccessClaimHandler({ delegations }) {
+  /** @type {AccessClaimHandler} */
+  return async (invocation) => {
+    // @todo - this should filter based on invocation
```

Comment:
I will resolve this todo as part of https://github.com/web3-storage/w3protocol/issues/394
For now this implementation is sufficient for the most basic of 'test delegate, then claim' that I started.
But as that got larger, I realized I should just focus on `access/delegate` in this review round.